frontend/ui/widget/textboxwidget.lua

```diff
@@ -144,6 +163,15 @@ function TextBoxWidget:_splitCharWidthList()
             idx = idx + 1
             -- FIXME: reuse newline entry
             self.vertical_string_list[ln+1] = {text = "", offset = idx, width = 0}
+        else
+            -- If next char is a space, discard it so it does not become
+            -- an ugly leading space on this next line
+            if idx <= size and self.char_width_list[idx].char == " " then
+                -- It's probably not needed to append it to last line
+                -- (not sure if that's ok with cursor stuff)
+                -- self.vertical_string_list[ln].text = self.vertical_string_list[ln].text .. " "
```

Comment:
Previously, you were either appending this space to this line, or prepending it to next line.
With my aditional checks, we'll be there with this space that we split/wrap lines on.
I think it makes no difference appending this space to current line, or ignoring it, except for the following:
- if we append it, should we increase `self.vertical_string_list[ln].width` with this space's size for the sake of coherency ? even if we splitted there so width does not exceed a limit.
- if we append it, we will render it, and this space may eat some bit of the scroll bar (depends on font, and there is there this HorizontalSpan that will make that ok)
- again, i dont know about cursor stuff, I don't know where there is an editable textboxwidget to test with

Also, when meeting a hard newline, you don't add it to any vertical_string_list[ln].text, you just use it to do the split. I think it's the same for this space at this point in the code.
So, i think just saying that in a comment is enough. Should i add what i just wrote to that comment, if you agree with it ?